torchcfm/conditional_flow_matching.py

```diff
@@ -47,21 +48,14 @@ class ConditionalFlowMatcher:
     - score function $\nabla log p_t(x|x0, x1)$
     """
 
-    def __init__(self, sigma: float = 0.0):
+    def __init__(self, sigma: Union[float, int] = 0.0):
         r"""Initialize the ConditionalFlowMatcher class. It requires the hyper-parameter $\sigma$.
 
         Parameters
         ----------
-        sigma : float
+        sigma : Union[float, int]
         """
-        self._sigma = sigma
-
-    @property
```

Comment:
Looks good. One comment - the `@property` method is useful for allowing one to "get" a variable, without changing it. So if someone did

```
obj = FlowMatcher(sigma=1.0)
obj.sigma = 2
```
they would get an error. They **could** be doing something like `obj._sigma = 2` to overwrite the property, but they would be very intentionally doing this. So it's mostly a way of telling the user how to now shoot themselves in the foot.

If you think it would be bad practice for people to be overwriting the `obj.sigma` property, then adding the `@property` logic back in still makes sense (sans the type casting).

Reply:
Thank you @josephdviviano for the clarification! I am not sure what is the best for sigma, I think it is fine for now if we keep it like this, but happy to make the change if someone thinks otherwise.